### DISCUSSION TITLE
AuthService set up error Expected '{' in class

### DISCUSSION THREAD

**Discussion by bradjeancy** [community/community#144589] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hi,

So I am trying to set up my authServices to fetch and update user data from my app to the firebase database but I am getting this error even though I also have a } at the end of this code further down below. can anyone help me out and let me know know how to fix?

![2024-11-15_01-55-31](https://github.com/user-attachments/assets/5c306d70-2194-4521-9c87-0db4c85ea7b7)

**Comment by Limzen** [comment]
The error message "Expected '{' in class" in your `AuthViewModel` class is likely due to a typo or syntax error. From the provided screenshot, the issue appears to be with how the `ObservableObject` is being declared. 

In Swift, the correct declaration of an observable object is:

```swift
class AuthViewModel: ObservableObject {
```

In the screenshot, it looks like there might be a typo with a space in `Observable Object`. It should be `ObservableObject` without any space.

Ensure the line reads:

```swift
class AuthViewModel: ObservableObject {
```

Make this correction and see if the error persists.

**Reply by bradjeancy** [reply]
Thank you

**Comment by BancoCentraI** [comment]
A mensagem de erro "Esperado '{' na classe" na sua classe AuthViewModel provavelmente é causada por um erro de digitação ou sintaxe. Pelo que foi mostrado na captura de tela, o problema parece estar na declaração do ObservableObject.

Em Swift, a maneira correta de declarar um objeto observável é:

swift
Copy code
class AuthViewModel: ObservableObject {
Na captura de tela, pode ser que tenha ocorrido um erro ao escrever "ObservableObject" com um espaço extra, como "Observable Object". O correto é "ObservableObject", sem nenhum espaço.

Certifique-se de que a linha esteja escrita assim:

swift
Copy code
class AuthViewModel: ObservableObject {

**Reply by bradjeancy** [reply]
Thank you
